Source/DafnyLanguageServer.Test/Lookup/HoverVerificationTest.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.Dafny.LanguageServer.IntegrationTest.Extensions;
+using Microsoft.Dafny.LanguageServer.IntegrationTest.Util;
+using Microsoft.Dafny.LanguageServer.Language;
+using Microsoft.Dafny.LanguageServer.Workspace;
+using Microsoft.Dafny.LanguageServer.Workspace.Notifications;
+using Microsoft.Extensions.Configuration;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.Dafny.LanguageServer.IntegrationTest.Lookup {
+  [TestClass]
+  public class HoverVerificationTest : DafnyLanguageServerTestBase {
+    private const int MaxTestExecutionTimeMs = 10000;
+
+    private ILanguageClient client;
+    private TestNotificationReceiver<CompilationStatusParams> notificationReceiver;
+    private IDictionary<string, string> configuration;
+
+    [TestInitialize]
+    public Task SetUp() => SetUp(null);
+
+    public async Task SetUp(IDictionary<string, string> configuration) {
+      this.configuration = configuration;
+      notificationReceiver = new();
+      client = await InitializeClient(options => {
+        options
+          .AddHandler(DafnyRequestNames.CompilationStatus, NotificationHandler.For<CompilationStatusParams>(notificationReceiver.NotificationReceived));
+      });
+    }
+
+    protected override IConfiguration CreateConfiguration() {
+      return configuration == null
+        ? base.CreateConfiguration()
+        : new ConfigurationBuilder().AddInMemoryCollection(configuration).Build();
+    }
+
+    [TestMethod, Timeout(MaxTestExecutionTimeMs)]
+    public async Task HoverGetsBasicAssertionInformation() {
+      var documentItem = await GetDocumentItem(@"
+method Abs(x: int) returns (y: int)
+//     ^ Hover #4
+    ensures y >= 0
+{ //           ^ hover #1
+  if x < 2 { // Hover #2 on the brace
+    return -x;
+  }
+  assert x > 2; // Hover #3 on the '>'
+  return x;
+}
+", CompilationStatus.VerificationFailed);
+      // When hovering the postcondition, it should display the position of the failing path
+      await AssertHoverMatches(documentItem, (2, 15),
+        @"assertion #1/2 of [batch](???) #1/1 checked in ???ms with ??? resource count:  
+`testFile0.dfy(6, 5): `*A postcondition might not hold on this return path.*"
+      );
+      // When hovering the failing path, it does not display the position of the failing postcondition
+      // because the IDE extension already does it.
+      await AssertHoverMatches(documentItem, (5, 4),
+        @"assertion #1/2 of [batch](???) #1/1 checked in ???ms with ??? resource count:  
```

Comment:
Question to reviewers: should I display `batch #1/1` when there is only one batch?
It's useful to have the word batch so that the link is available, but I could as well put the link on assertion and remove `of batch #1/1`. I currently favored consistency with other messages.

Reply:
Yeah, I think it's reasonable to always say what batch it's in, partly because it clarifies that the time and resource counts listed are for batches, not individual assertions (and only apply to individual assertions if they're 1/1 in the current batch).